app/controllers/planning_application/site_notices_controller.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+class PlanningApplication
+  class SiteNoticesController < AuthenticationController
+    include ActionView::Helpers::SanitizeHelper
+
+    before_action :set_planning_application
+    before_action :set_site_notice, except: %i[new create]
+    before_action :ensure_public_portal_is_active, only: :create
+
+    def show
+      respond_to do |format|
+        format.html
+      end
+    end
+
+    def new
+      @site_notice = SiteNotice.new
+    end
+
+    def edit
+      respond_to do |format|
+        format.html
+      end
+    end
+
+    def create
```

Comment:
Possibly a lot happening and with several db updates in this controller action so thinking if we could maybe simplify and move things to the model, especially the email sending / auditing?
I'll have a go locally